tests/test.rfb.js

```diff
@@ -3125,28 +3125,24 @@ describe('Remote Frame Buffer Protocol Client', function () {
 
                     sendFbuMsg([{ x: 0, y: 0, width: 0, height: 0, encoding: -261 }], [data], client);
 
+                    expect(client._remoteCapsLock).to.equal(true);
+                    expect(client._remoteNumLock).to.equal(true);
+
                     data = [];
                     push8(data, 0b011);
 
                     sendFbuMsg([{ x: 0, y: 0, width: 0, height: 0, encoding: -261 }], [data], client);
 
+                    expect(client._remoteCapsLock).to.equal(false);
+                    expect(client._remoteNumLock).to.equal(true);
+
                     data = [];
                     push8(data, 0b000);
 
                     sendFbuMsg([{ x: 0, y: 0, width: 0, height: 0, encoding: -261 }], [data], client);
 
-                    expect(spy).to.have.been.calledThrice;
-                    expect(spy.args[0][0].detail.capsLock).to.equal(true);
-                    expect(spy.args[0][0].detail.numLock).to.equal(true);
-                    expect(spy.args[0][0].detail.scrollLock).to.equal(false);
-
-                    expect(spy.args[1][0].detail.capsLock).to.equal(false);
-                    expect(spy.args[1][0].detail.numLock).to.equal(true);
-                    expect(spy.args[1][0].detail.scrollLock).to.equal(true);
-
-                    expect(spy.args[2][0].detail.capsLock).to.equal(false);
-                    expect(spy.args[2][0].detail.numLock).to.equal(false);
-                    expect(spy.args[2][0].detail.scrollLock).to.equal(false);
+                    expect(client._remoteCapsLock).to.equal(false);
+                    expect(client._remoteNumLock).to.equal(false);
```

Comment:
We want to avoid poking around in internal variables as that makes the tests very fragile.

Do you think you could adjust the tests so that they only look at external behaviour? I.e. that it only sends the fake key presses when supported or needed.